guid  must be specified in episode's .yml file

sockeqwe opened the pull request:
What I changed?
- `guid` must be specified in each episode in `.yml` file for each manually.
- Added `episode.title` to error messages of `EpisodeValidator` so that we know which episode caused the validation error.

ming13 commented on src/main/kotlin/io/thecontext/ci/validation/PodcastValidator.kt:
```diff
@@ -24,7 +24,7 @@ class PodcastValidator(
                     Single.fromCallable {
                         if (people.find { it.id == personId } == null) {
-                            ValidationResult.Failure("Person [$personId] is not defined.")
+                            ValidationResult.Failure("Podcast Authors or Owner: Person [$personId] is not defined.")
```
`Podcast person [$personId]`

ming13 commented on src/main/kotlin/io/thecontext/ci/value/Episode.kt:
```diff
@@ -1,11 +1,15 @@
 package io.thecontext.ci.value
 
 import com.fasterxml.jackson.annotation.JsonProperty
+import java.io.File
```
?

ming13 commented on src/test/kotlin/io/thecontext/ci/Data.kt:
```diff
@@ -34,6 +34,7 @@ val testPodcast = Podcast(
 )
 
 val testEpisode = Episode(
+        guid = "TestEpisode",
```
`Episode ID`

`Test *` just adds clutter to error messages because everything is being prepended with `Test`.

sockeqwe replied:
👍

ming13 commented on src/test/kotlin/io/thecontext/ci/validation/EpisodeValidatorSpec.kt:
```diff
@@ -13,6 +13,30 @@ class EpisodeValidatorSpec {
     init {
         val env by memoized { Environment() }
 
+        context("guid validation failed") {
```
`\n`

ming13 commented on src/test/kotlin/io/thecontext/ci/validation/EpisodeValidatorSpec.kt:
```diff
@@ -13,6 +13,30 @@ class EpisodeValidatorSpec {
 
+        context("guid validation failed") {
+            beforeEach {
+                env.urlValidator.result = ValidationResult.Failure("nope")
```
Tests are successful only because URL validation fails first.

sockeqwe commented on src/main/kotlin/io/thecontext/ci/output/PodcastXmlFormatter.kt:
```diff
@@ -46,6 +46,7 @@ interface PodcastXmlFormatter {
                             "build_date" to LocalDate.now().toRfc2822(),
                             "episodes" to episodes.map { (episode, episodeMarkdown) ->
                                 mapOf(
+                                        "guid" to episode.guid,
```
Why would you suggest to rename it to `id`? Just curious ...

ming13 replied:
It is simpler and carries the same meaning. Besides, GUID is kind of an implementation detail.

sockeqwe replied:
ok, renamed it.

sockeqwe commented on src/main/kotlin/io/thecontext/ci/validation/EpisodeValidator.kt:
```diff
@@ -16,28 +16,49 @@ class EpisodeValidator(
+            }
+
+            if (value.guid.contains(' ')) {
+                ValidationResult.Failure("$episodeIdentifierForError: guid contains white space")
```
👍 I thought whitespaces should not be part of guid. Will remove it.

sockeqwe commented on src/main/kotlin/io/thecontext/ci/value/Episode.kt:
```diff
@@ -1,11 +1,15 @@
         val slug: String = "",
 
+        @JsonProperty("guid")
+        val guid : String,
```
I'm fine with renaming it, I'm just wondering why `guid` is not an option? I mean, it's called `guid` in rss feed specification ...

sockeqwe commented on src/main/kotlin/io/thecontext/ci/validation/EpisodeValidator.kt:
```diff
@@ -16,28 +16,49 @@ class EpisodeValidator(
+                ValidationResult.Failure("$episodeIdentifierForError: guid contains white space")
+            }
+
+            ValidationResult.Success
```
🤦‍♂️

sockeqwe commented:
@ming13 I have renamed `guid` to `id`.

Furthermore, I have added a `EpisodeListValidator` that checks if the id's of our episodes are unique with our lists (I'm sorry for not having put this in a dedicated pull request, but I'm too lazy to resolve foreseeable merge conflicts given the fact that `master` branch currently doesn't have `episode.id` at all. I hope you don't mind)
